include/CL/sycl/vec.hpp

```diff
@@ -143,6 +143,72 @@ class vec : public detail::small_array<DataType,
     return std::tuple_cat(flatten<V>(i)...);
   }
 
+public:
+  /** An accessor to the first variable of a small array
+   */
+  DataType& x(){
+    static_assert(NumElements >= 1, "can't access to small_array[0] if NumElements < 1");
+    return (*this)[0];
+  }
+
+
+  /** An accessor to the second variable of a small array
+  */
+  DataType& y(){
+    static_assert(NumElements >= 2, "can't access to small_array[1] if NumElements < 2");
+    return (*this)[1];
+  }
+
+
+  /** An accessor to the third variable of a small array
+  */
+  DataType& z(){
+    static_assert(NumElements >= 3, "can't access to small_array[2] if NumElements < 3");
+    return (*this)[2];
+  }
+
+  /** An accessor to the third variable of a small array
+  */
+  DataType& w(){
+    static_assert(NumElements >= 4, "can't access to small_array[3] if NumElements < 4");
+    return (*this)[3];
+  }
+
+  /** Accessors to access hex indexed members of a small array.
+   * There are two macros, one for 0-9, one for A-F.
+   */
+#define TRISYCL_DECLARE_S(x)							\
+  DataType& s##x() {							\
+    static_assert(NumElements >= (x + 1), "can't access to small_array["#x"] if NumElements < "#x"+1"); \
+    return (*this)[(x)];						\
+  }
+
+#define TRISYCL_DECLARE_Sx(x)							\
+  DataType& s##x() {							\
+    static_assert(NumElements >= (0x##x + 1), "can't access to small_array[0x"#x"] if NumElements < 0x"#x"+1"); \
+    return (*this)[(0x##x)];						\
+  }
+
+  TRISYCL_DECLARE_S(0);
+  TRISYCL_DECLARE_S(1);
+  TRISYCL_DECLARE_S(2);
+  TRISYCL_DECLARE_S(3);
+  TRISYCL_DECLARE_S(4);
+  TRISYCL_DECLARE_S(5);
+  TRISYCL_DECLARE_S(6);
+  TRISYCL_DECLARE_S(7);
+  TRISYCL_DECLARE_S(8);
+  TRISYCL_DECLARE_S(9);
+  TRISYCL_DECLARE_Sx(A);
+  TRISYCL_DECLARE_Sx(B);
+  TRISYCL_DECLARE_Sx(C);
+  TRISYCL_DECLARE_Sx(D);
+  TRISYCL_DECLARE_Sx(E);
+  TRISYCL_DECLARE_Sx(F);
+
+#undef TRISYCL_DECLARE_S
+#undef TRISYCL_DECLARE_Sx
+
 
   /// \todo To implement
 #if 0
```

Comment:
I have the feeling that some of the following are implemented now, isn't it?

Reply:
Yeah some of those are, I'll follow up later with some removal of that code.